### DISCUSSION TITLE
Still getting build error regarding the 2016 agent

### DISCUSSION THREAD

**Discussion by theikell** [actions/virtual-environments#6110] [General questions] [answered]
I have my Agent pool set to Azure Pipelines and Agent Specification set to windows-2019 everywhere that I can find it to be set, and yet I'm still getting the "The Windows 2016 environment is deprecated, consider switching to windows-2019 or windows-2022 (windows-latest)." message, and I can't figure out why. Any suggestions would be very much appreciated.

**Comment by miketimofeev** [comment]
@theikell could you share a screenshot of your pipeline configuration? There are two places where the agent can be set:
![image](https://user-images.githubusercontent.com/48208649/185971304-76dbbd3e-40cb-4262-826d-f96e72009aea.png)

**Comment by theikell** [comment]
@miketimofeev this is my config
![image](https://user-images.githubusercontent.com/5521596/185972777-bf70b85c-a53b-4563-b979-2a82d1723e56.png)

**Comment by miketimofeev** [comment, marked as answer]
@theikell and what agent is selected in the:
```
Phase 1
Run on agent
```

**Reply by theikell** [reply]
I think I looked everywhere else but there. I incorrectly expected it would inherit the settings from the parent. That fixed it, thanks a ton!
